subsys/logging/log_core.c

```diff
@@ -256,14 +248,17 @@ void log_init(void)
 
 	/* Assign ids to backends. */
 	for (i = 0; i < log_backend_count_get(); i++) {
-		log_backend_id_set(log_backend_get(i),
+		const struct log_backend *backend = log_backend_get(i);
+
+		log_backend_id_set(backend,
 				   i + LOG_FILTER_FIRST_BACKEND_SLOT_IDX);
-	}
 
-	if (IS_ENABLED(CONFIG_LOG_BACKEND_UART)) {
-		backend_filter_init(uart_backend);
-		log_backend_uart_init();
-		log_backend_activate(uart_backend, NULL);
+		backend_filter_init(backend);
+		if (backend->api->init) {
+			backend->api->init();
+		}
+
+		log_backend_activate(backend, NULL);
```

Comment:
There will be use cases where we don't want to enable all backends. Example: log backend over bluetooth where backend will be registered but not activated until connection is established.

Reply:
Is the expectation then, that backends which are meant to be started if configured, be called from here (like the UART one was, but the others be activated from other places?

Reply:
yes, i was thinking the same that we might want to extend backend registeration macro to has something autostart flag.

Reply:
Lets keep this change as is and i will try to address that in separate PR.

Reply:
> Lets keep this change as is and i will try to address that in separate PR.

That would be best yes.

>  extend backend registeration macro to has something autostart flag.

Note that would use memory, while keeping that information implicitly in the order would not.

Reply:
I think we might handle some special cases in api->init()